src/views/PasswordReset.vue

```diff
@@ -0,0 +1,150 @@
+                single-line
+                solo
+                :rules="[
+                  password.rules,
+                  passwordConfirmationRules(
+                    password.value,
+                    confirmedPassword.value
+                  )
+                ]"
+                :type="passwordFieldType"
+                :append-icon="visibilityIcon"
+                @click:append="switchVisibility()"
+              ></v-text-field>
+              <v-text-field
+                label="Confirm password"
+                v-model="confirmedPassword.value"
+                single-line
+                solo
+                :rules="[
+                  confirmedPassword.rules,
+                  passwordConfirmationRules(
+                    password.value,
+                    confirmedPassword.value
+                  )
+                ]"
+                :type="passwordFieldType"
+                :append-icon="visibilityIcon"
+                @click:append="switchVisibility()"
+              ></v-text-field></v-form
+            ><v-layout column align-center>
+              <v-flex>
+                <v-btn
+                  color="primary"
+                  class="mb-3"
+                  :disabled="!isValid"
+                  @click="changePassword()"
+                  >Change password</v-btn
+                >
+              </v-flex></v-layout
+            >
+          </div>
+          <v-layout column align-center
+            ><v-flex
+              ><v-progress-circular
+                v-if="isRequestingPasswordChange"
+                indeterminate
+                color="primary"
+                size="170"
+              ></v-progress-circular></v-flex></v-layout></v-card
+      ></v-flex>
+      <v-snackbar color="error" v-model="isTokenInvalid" :timeout="9000" bottom>
+        This link is no longer valid. Please request another password reset
+        email.
+      </v-snackbar>
+    </v-layout>
+  </v-container>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import axios from "axios";
+import { AxiosResponse } from "axios";
+import * as settings from "@/utils/settings";
+
+export default Vue.extend({
+  name: "PasswordReset",
+  data: () => ({
+    token: null,
+    isTokenInvalid: false,
+    isValid: true,
+    passwordFieldType: "password",
+    visibilityIcon: "visibility",
+    isRequestingPasswordChange: false,
+    password: {
+      value: null,
+      rules: v => {
+        if (!v) {
+          return "Enter your password";
+        }
+        if (v.length < 8) {
+          return "Password should be at least 8 characters long";
+        }
+        return true;
+      }
+    },
+    confirmedPassword: {
+      value: null,
+      rules: v => {
+        if (!v) {
+          return "Confirm your password";
+        }
+        return true;
+      }
+    }
+  }),
+  created() {
+    this.token = this.$route.params.token;
+    // Verify token validity
```

Comment:
Ah, I see now what this endpoint was used for. The expiry is readable by the client though so we don't need to make a request to check it